src/parser/jobs/ast/modules/Lightspeed.tsx

```diff
@@ -0,0 +1,231 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {RotationTable} from 'components/ui/RotationTable'
+import {Status} from 'data/STATUSES'
+import {Event, Events} from 'event'
+import _ from 'lodash'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React, {Fragment} from 'react'
+import {Icon} from 'semantic-ui-react'
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+
+const BASE_GCDS_PER_WINDOW = 6
+
+class LIGHTSPEED_Window {
+	start: number
+	end?: number
+
+	rotation: Array<Events['action']> = []
+	gcdCount: number = 0
+	mpSavings: number = 0
+	trailingGcdEvent?: Events['action']
+
+	buffsRemoved: Array<Status['id']> = []
+	deathTruncated: boolean = false
+
+	constructor(start: number) {
+		this.start = start
+	}
+}
+
+// in this module we only want to track Lightspeed windows opened by
+// the character selected for analysis. windows that clip into
+// AST Lightspeed will be marked.
+// Used DNC Technicalities as basis for this module. Rewritten from previous module for consistency purposes
+export default class Lightspeed extends Analyser {
+	static override handle = 'Lightspeed'
+	static override title = t('ast.Lightspeed.title')`Lightspeed`
+	static override displayOrder = DISPLAY_ORDER.LIGHTSPEED
+
+	@dependency private timeline!: Timeline
+	@dependency private data!: Data
+
+	private history: LIGHTSPEED_Window[] = []
+
+	override initialise() {
+		const lightspeedFilter = filter<Event>().status(this.data.statuses.LIGHTSPEED.id)
+
+		this.addEventHook(lightspeedFilter.type('statusApply')
+			.target(this.parser.actor.id), this.tryOpenWindow)
+		this.addEventHook(lightspeedFilter.type('statusRemove')
+			.target(this.parser.actor.id), this.tryCloseWindow)
+
+		this.addEventHook(filter<Event>().source(this.parser.actor.id).type('action'), this.onCast)
+	}
+
+	private tryOpenWindow(event: Events['statusApply']): LIGHTSPEED_Window {
+		const lastWindow: LIGHTSPEED_Window | undefined = _.last(this.history)
+
+		// If that happens, re-open the last window and keep tracking
+		if (lastWindow != null) {
+			if (!lastWindow.end) {
```

Comment:
End isn't a boolean. Prefer to explicitly null check this. I think there's an easier way to write this whole block but it's not coming to me atm.

Reply:
it has been updated to be simpler, since it always returned the same thing as the end, I just removed whatever we were originally looking for anyway which was if end exists, we undefine, otherwise return